quarkus/runtime/src/test/java/org/keycloak/quarkus/runtime/configuration/test/ConfigurationTest.java

```diff
@@ -564,20 +584,59 @@ public void testKeystoreConfigSourcePropertyMapping() {
                 .addDiscoveredSources()
                 .build();
 
-        assertEquals(config.getConfigValue("smallrye.config.source.keystore.kc-default.password").getValue(),config.getConfigValue("kc.config-keystore-password").getValue());
+        assertEquals(config.getConfigValue("smallrye.config.source.keystore.kc-default.password").getValue(), config.getConfigValue("kc.config-keystore-password").getValue());
         // Properties are loaded from the file - secret can be obtained only if the mapping works correctly
         ConfigValue secret = config.getConfigValue("my.secret");
         assertEquals("secret", secret.getValue());
     }
 
+    @Test
+    public void testNotMappedQuarkusPropertyCli() {
+        setCliArgs("--log=console", "--log-file=something");
+        SmallRyeConfig config = createConfig();
+
+        var kcProperty = config.getConfigValue("kc.log-file");
+        assertThat(kcProperty.getValue(), nullValue());
+
+        var quarkusProperty = config.getConfigValue("quarkus.log.file.path");
+        assertThat(quarkusProperty.getValue(), not(equalTo("something")));
+
+        //specified in test keycloak.conf
+        assertThat(quarkusProperty.getValue(), not(equalTo("random/path")));
+    }
+
+    @Test
+    public void testNotMappedQuarkusPropertyEnv() {
+        putEnvVar("KC_LOG", "console");
+        putEnvVar("KC_LOG_FILE", "something-env");
+        putEnvVar("QUARKUS_LOG_FILE_PATH", "random/env/path");
+
+        SmallRyeConfig config = createConfig();
+
+        var kcProperty = config.getConfigValue("kc.log-file");
+        assertThat("Config source: " + kcProperty, kcProperty.getValue(), nullValue());
+        // the disabled property is not reflected in configuration, but env var still present
+        assertThat(getEnvVar("KC_LOG_FILE"), equalTo("something-env"));
+
+        var quarkusProperty = config.getConfigValue("quarkus.log.file.path");
+        assertThat(quarkusProperty.getValue(), not(equalTo("something-env")));
+        assertThat(quarkusProperty.getValue(), equalTo("random/env/path"));
+    }
+
+    private static void setCliArgs(String... args) {
+        System.setProperty(CLI_ARGS, String.join(ARG_SEPARATOR, args));
+    }
+
     private Config.Scope initConfig(String... scope) {
         Config.init(new MicroProfileConfigProvider(createConfig()));
         return Config.scope(scope);
     }
 
     private SmallRyeConfig createConfig() {
         KeycloakConfigSourceProvider.reload();
+        KeycloakConfigSourceProvider.sanitizeConfigSources();
         ConfigProviderResolver.setInstance(null);
-        return ConfigUtils.configBuilder(true, LaunchMode.NORMAL).build();
+        // reflect real config source configuration
+        return new KeycloakConfigSourceProvider().configBuilder(ConfigUtils.emptyConfigBuilder()).build();
```

Comment:
Isn't KeycloakConfigSourceProvider picked up via the ServiceLoader? Or do you see a case where it is not?

Reply:
The thing is that I tried to get rid of the default `EnvConfigSource`, as we don't want to have processed Keycloak ENV vars there. For that purpose, I've created custom `SmallRyeEnvConfigSource`[1].

The reason behind that is to not process values provided via disabled Keycloak ENV vars.

But if I'm not mistaken, the `EnvConfigSource` is provided into the default config sources anyway through the SmallRyeConfigProviderResolver, and its loading via the Service loader. It means, the test case for the ENV var is working only in the test class.

But I'm not sure, if I should spend more time to have the ability to ignore Keycloak ENV vars when their mappers are disabled.

@shawkins WDYT? 

[1] https://github.com/keycloak/keycloak/pull/25333/files/323aff68e704674e2cd9fa33910d07428f744fdb#diff-3574824ab9fd51968b78e779c88ff3b7ea5688a840dbd57e285518d236b84094R37